### PR TITLE
Support multiple road classes in `RouteOptions.roadClassesToAvoid` and `RouteOptions.roadClassesToAllow` properties

### PR DESCRIPTION
**Pull request by azarovalex**
[Directions server-side API](https://docs.mapbox.com/api/navigation/directions/) now allows passing multiple road classes in these fields, so the corresponding SDK code is updated.

Looks like `RoadClasses.tunnel` and `RoadClasses.restricted` are not supported by Directions API in `RouteOptions.roadClassesToAvoid` anymore, so I updated documentation for these properties.

Error when trying to avoid tunnels:
image

@1ec5 Looks like `RoadClasses.tunnel` and `RoadClasses.restricted` were only used in `RouteOptions.roadClassesToAvoid `, now that Directions API doesn't support them, should we mark these properties as deprecated?

### CODE REVIEW

File: Sources/MapboxDirections/RoadClasses.swift [hunk 1]
```diff
@@ -185,3 +185,13 @@ extension RoadClasses: Codable {
         }
     }
 }
+
+extension RoadClasses {
+    static var validAvoidClasses: RoadClasses {
+        return [.toll, .motorway, .ferry, .unpaved, .cashTollOnly]
```
Comment (1ec5) [inline Sources/MapboxDirections/RoadClasses.swift:191]: I’m wary of hard-coding these allowlists. We’ve done that in the past and it has gotten outdated. Better to document our current expectations and allow the request to return an error.

File: Sources/MapboxDirections/RouteOptions.swift [hunk 2]
```diff
@@ -281,20 +281,21 @@ open class RouteOptions: DirectionsOptions {
...
-        
-        if !roadClassesToAvoid.isEmpty && roadClassesToAvoid.isDisjoint(with: [.highOccupancyVehicle2, .highOccupancyVehicle3, .highOccupancyToll]) {
-            let allRoadClasses = roadClassesToAvoid.description.components(separatedBy: ",").filter { !$0.isEmpty }
-            precondition(allRoadClasses.count < 2, "You can only avoid one road class at a time.")
-            if let firstRoadClass = allRoadClasses.first {
-                params.append(URLQueryItem(name: CodingKeys.roadClassesToAvoid.stringValue, value: firstRoadClass))
+
+        if !roadClassesToAvoid.isEmpty {
+            if !roadClassesToAvoid.isSubset(of: .validAvoidClasses) {
+                assertionFailure("Provided road classes cannot be avoided")
```
Comment (1ec5) [inline Sources/MapboxDirections/RouteOptions.swift:287]: In general, we avoid client-side validation, except for options that don’t make logical sense.

File: Sources/MapboxDirections/RoadClasses.swift (hunk 1 above)
Comment (azarovalex) [reply Sources/MapboxDirections/RoadClasses.swift]: Ok, I removed the hardcode and filtering of allowed road classes, now the server will perform the check

File: Sources/MapboxDirections/RouteOptions.swift (hunk 2 above)
Comment (azarovalex) [reply Sources/MapboxDirections/RouteOptions.swift]: Agree, removed the assertion.

**Review by bamx23 — APPROVED** [review]

**Pull request merged**